pkg/cmd/util/ip.go

```diff
@@ -59,3 +64,53 @@ func AllLocalIP4() ([]net.IP, error) {
 	}
 	return ret, nil
 }
+
+// Validate given node IP belongs to the current host
+func ValidateNodeIP(nodeIP net.IP) error {
+	if nodeIP.To4() == nil && nodeIP.To16() == nil {
+		return fmt.Errorf("nodeIP must be a valid IP address")
+	}
+	if nodeIP.IsLoopback() {
+		return fmt.Errorf("nodeIP can't be loopback address")
+	}
+	if nodeIP.IsMulticast() {
+		return fmt.Errorf("nodeIP can't be a multicast address")
+	}
+	if nodeIP.IsLinkLocalUnicast() {
+		return fmt.Errorf("nodeIP can't be a link-local unicast address")
+	}
+	if nodeIP.IsUnspecified() {
+		return fmt.Errorf("nodeIP can't be an all zeros address")
```

Comment:
how about "nodeIP must be specified" or "nodeIP must not equal 0.0.0.0"

Reply:
This method was direct copy+paste from kubernetes repo. I did not want to import the corresponding k8s pkg as that will increase the binary from 32 MB to 40+ MB (also needs approval). 
Any way, Fixed.

Reply:
Didn't realize this was copied — no objection to keeping it the same as upstream to ease diffing

Reply:
> This method was direct copy+paste from kubernetes repo. I did not want to import the corresponding k8s pkg as that will increase the binary from 32 MB to 40+ MB (also needs approval).

Ah, if this is copy-and-paste of the upstream logic, and we have a reason for copying and pasting instead of importing, that largely addresses my earlier concerns.  Maybe add a comment referencing the upstream code and explaining why we reproduce it?

Reply:
Okay, reverted to the original k8s code. Added a comment on why this was copied.